File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
import sys
from collections import deque

sys.setrecursionlimit(10**6)
input = sys.stdin.readline

N , M , V = map(int, input().split())

graph = [[] for _ in range(N+1)]
visited_dfs = [0]*(N+1)
visited_bfs = [0]*(N+1)



for _ in range(M):
    a, b = map(int, input().split())
    graph[a].append(b)
    graph[b].append(a)
    
for x in range(1,N+1):
    graph[x].sort()
    
def recur(node):

    
    visited_dfs[node] = 1
    print(node, end=' ')
    
    
    for nxt in graph[node]:
        if visited_dfs[nxt] == 0:
            recur(nxt)
    
   
def bfs(V):
    q = deque()
    q.append(V)
    visited_bfs[V] = 1
    
    
    while q:
        
        node = q.popleft()
        
       
        print(node, end=' ')
        
        
        for nxt in graph[node]:
            if visited_bfs[nxt] == 0:
                visited_bfs[nxt] = 1
                q.append(nxt)
            

        


        
  
recur(V)
print()
bfs(V)

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.cc
#include <iostream>
#include <vector>
#include <tuple>
#include <set>
#include <map>
#include <string>
#include <algorithm>
#include <iomanip>
#include <functional>
#include <cctype>
#include <queue>

const int MAX = 1001;

std::vector<int> graph[MAX];
bool visited_dfs[MAX];
bool visited_bfs[MAX];

int N, M, V;

void dfs(int node) {
    visited_dfs[node] = true;
    std::cout << node << ' ';

    for (int nxt : graph[node]){
        if (!visited_dfs[nxt]){
            dfs(nxt);
        }
    }
}

void bfs(int start) {
    std::queue<int> q;
    q.push(start);
    visited_bfs[start] = true;

    while (!q.empty()) {
        int node = q.front();
        q.pop();
        std::cout << node << ' ';

        for (int nxt : graph[node]){
            if (!visited_bfs[nxt]) {
                visited_bfs[nxt] = true;
                q.push(nxt);
            }
        }
    }
}
int main() {
    std::ios::sync_with_stdio(false);
    std::cin.tie(0);

    std::cin >> N >>M >> V;

    for (int i=0; i<M; i++) {
        int a,b;
        std::cin >> a >> b;
        graph[a].push_back(b);
        graph[b].push_back(a);
    }

    for (int i=1; i<=N; i++) {
        std::sort(graph[i].begin(), graph[i].end());
    }

    dfs(V);
    std::cout << '\n';
    bfs(V);
    std::cout << '\n';

    return 0;
}